app.js

```diff
@@ -7,15 +7,58 @@ client.on('ready', () => {
 });
 client.login(process.env.BOT_TOKEN)
 
+async function setup(msg) {
+    //https://greenwichmeantime.com/time-zone/definition/
+    //https://github.com/ac360/currency-codes-array-ISO4217
+    const timezoneCodes = ['IDLW', 'NT', 'HST', 'AKST', 'PST', 'MST', 'CST', 'EST', 'AST', 'ART', 'AT', 'WAT', 'GMT', 'CET', 'EET', 'MSK', 'AMT', 'PKT', 'OMSK', 'KRAT', 'CST', 'JST', 'AEST', 'SAKT', 'NZST'];
+    const currencyCodes = ['AED', 'AFN', 'ALL', 'AMD', 'ANG', 'AOA', 'ARS', 'AUD', 'AWG', 'AZN', 'BAM', 'BBD', 'BDT', 'BGN', 'BHD', 'BIF', 'BMD', 'BND', 'BOB', 'BOV', 'BRL', 'BSD', 'BTN', 'BWP', 'BYR', 'BZD', 'CAD', 'CDF', 'CHE', 'CHF', 'CHW', 'CLF', 'CLP', 'CNY', 'COP', 'COU', 'CRC', 'CUC', 'CUP', 'CVE', 'CZK', 'DJF', 'DKK', 'DOP', 'DZD', 'EGP', 'ERN', 'ETB', 'EUR', 'FJD', 'FKP', 'GBP', 'GEL', 'GHS', 'GIP', 'GMD', 'GNF', 'GTQ', 'GYD', 'HKD', 'HNL', 'HRK', 'HTG', 'HUF', 'IDR', 'ILS', 'INR', 'IQD', 'IRR', 'ISK', 'JMD', 'JOD', 'JPY', 'KES', 'KGS', 'KHR', 'KMF', 'KPW', 'KRW', 'KWD', 'KYD', 'KZT', 'LAK', 'LBP', 'LKR', 'LRD', 'LSL', 'LTL', 'LVL', 'LYD', 'MAD', 'MDL', 'MGA', 'MKD', 'MMK', 'MNT', 'MOP', 'MRO', 'MUR', 'MVR', 'MWK', 'MXN', 'MXV', 'MYR', 'MZN', 'NAD', 'NGN', 'NIO', 'NOK', 'NPR', 'NZD', 'OMR', 'PAB', 'PEN', 'PGK', 'PHP', 'PKR', 'PLN', 'PYG', 'QAR', 'RON', 'RSD', 'RUB', 'RWF', 'SAR', 'SBD', 'SCR', 'SDG', 'SEK', 'SGD', 'SHP', 'SLL', 'SOS', 'SRD', 'SSP', 'STD', 'SYP', 'SZL', 'THB', 'TJS', 'TMT', 'TND', 'TOP', 'TRY', 'TTD', 'TWD', 'TZS', 'UAH', 'UGX', 'USD', 'USN', 'USS', 'UYI', 'UYU', 'UZS', 'VEF', 'VND', 'VUV', 'WST', 'XAF', 'XAG', 'XAU', 'XBA', 'XBB', 'XBC', 'XBD', 'XCD', 'XDR', 'XFU', 'XOF', 'XPD', 'XPF', 'XPT', 'XTS', 'XXX', 'YER', 'ZAR', 'ZMW'];
+    const content = msg.content.slice(7).toUpperCase().trim().split(' ');  //7 is length of '!setup '
+    const timezone = content[0];
+    const currency = content[1];
+    let timezoneFound = false;
+    let currencyFound = false;
+    timezoneCodes.forEach(item => {
+        if(item === timezone){
+            timezoneFound = true;
+        }});
+    currencyCodes.forEach(item => {
+        if(item === currency){
+            currencyFound = true;
+        }});
+    if(timezoneFound && currencyFound){
+        msg.channel.send('Successful match for both timezone and currency!');
+        msg.channel.send('User: ' + msg.member.user.tag);
+        //msg.channel.send('Channel: ' + msg.channel.id); *this line for debugging not for display*
+    }
+    if(timezoneFound){
+        msg.channel.send('Timezone: ' + timezone);
+    }
+    else{
+        msg.channel.send('We could not find your timezone.');
+    }
+    if(currencyFound){
+        msg.channel.send('Currency: ' + currency);
+    }
+    else{
+        msg.channel.send('We could not find your currency.');
+    }
+    return {userId: msg.member.user.tag, channelId: msg.channel.id, timezone: this.timezone, currency: this.currency};
```

Comment:
no need for `this`, just `timezone` because its declared as a normal variable above

Reply:
Removed